lib/cn/move.c

```diff
@@ -109,13 +109,12 @@ cn_move(
         assert(list_empty(src_head));
         route_map_delete(tree->ct_route_map, src_node->tn_route_node);
         src_node->tn_route_node = NULL;
-    } else {
-        if (src_cnt > 0)
-            cn_tree_samp_update_move(w, src_node);
     }
 
-    if (src_cnt > 0)
-        cn_tree_samp_update_move(w, tgt_node);
+    cn_tree_samp(tree, &w->cw_samp_pre);
```

Comment:
Thanks for fixing this, Greg!

Is the samp update a no-op when joining an empty left node to a non-empty right node, i.e., when src_cnt == 0? The only thing that cn_move() does in this case is to delete 'src_node' from the route map.

IOW, would it make sense to put lines 114 to 117 under if (src_cnt > 0)?

Reply:
It's not exactly a no-op, because it zeroes out tn_samp when the node is empty, and hence it's reflects the fact that there are no kvsets.  Should we happen to use or look at tn_samp later it will correct.  We don't currently, but should we grow some code that does or someone looks at in the debugger it will be correct :-)